File: 1. Two Sum/Python/main.py
from typing import List

def main():
    sol = Solution()
    passed = 0
    total = 0

    try:
        with open("../input.txt", "r") as f:
            lines = f.readlines()
    except FileNotFoundError:
        print("Failed to open input.txt")
        return
    
    i = 0
    n = len(lines)

    while i < n:
        line = lines[i].strip()
        i += 1

        if not line:
            continue

        if len(line) < 2:
            break

        # Reading Array
        arr_str = line[1: -1]
        nums = []
        if arr_str:
            nums = list(map(int, arr_str.split(",")))

        # Reading Target
        if i >= n:
            break
        target = int(lines[i].strip())
        i += 1

        # Reading expected output
        if i >= n:
            break
        expected_line = lines[i].strip()
        i += 1

        expected_str = expected_line[1:-1]
        expected = []
        if expected_str:
            expected = list(map(int, expected_str.split(",")))

        total += 1
        result = sol.twoSum(nums, target)

        if result == expected:
            passed += 1
        else:
            print(f"Test Case {total}: FAILED")
            print(f"Input: nums = {nums}, target = {target}")
            print(f"Expected: {expected}")
            print(f"Your Output: {result}")
            return
    
    print(f"Result: {passed} / {total} test cases passed")


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(0, len(nums)):
            val = target - nums[i]
            if val in map:
                return [map[val], i]
            
            map[nums[i]] = i
        
        return []
    
if __name__ == "__main__":
    main()

File: 1. Two Sum/JavaScript/main.js
const fs = require("fs");

/** 
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
*/

var twoSum = function(nums, target){
    let map = new Map();
    for (let i = 0; i < nums.length; i++){
        let val = target - nums[i];
        if (map.has(val)){
            return [map.get(val), i];
        }
        map.set(nums[i], i);
    }
    return [];
}

let lines;

try{
    lines = fs.readFileSync("../input.txt", "utf-8").split(/\r?\n/);
}
catch (e){
    console.log("Failed to open input.txt");
    process.exit(1);
}

let passed = 0;
let total = 0;
let i = 0;

while (i < lines.length){
    let line = lines[i].trim();
    i++;

    if (!line){
        continue;
    }

    if (line.length < 2){
        break;
    }

    let arrStr = line.slice(1, -1);
    let nums = [];
    if (arrStr.length > 0){
        nums = arrStr.split(",").map(x => parseInt(x.trim()));
    }

    if (i >= lines.length){
        break;
    }
    
    let target = parseInt(lines[i].trim());
    i++;

    if (i >= lines.length){
        break;
    }

    let expectedLine = lines[i].trim();
    i++;

    let expectedStr = expectedLine.slice(1, -1);
    let expected = [];
    if (expectedStr.length > 0){
        expected = expectedStr.split(",").map(x => parseInt(x.trim()));
    }

    total++;
    let result = twoSum(nums, target);

    if (JSON.stringify(result) === JSON.stringify(expected)){
        passed++;
    }
    else{
        console.log(`Test Case ${total}: FAILED`);
        console.log(`Input: nums = [${nums}], target = ${target}`);
        console.log(`Expected: [${expected}]`);
        console.log(`Your Output: [${result}]`);
        process.exit(0);
    }
}

console.log(`Result: ${passed} / ${total} test cases passed`);